Dockerfile

```diff
@@ -208,17 +208,42 @@ RUN --mount=type=cache,target=/root/.cache/go-build \
         GOBIN=/build/ GO111MODULE=on go install "github.com/tc-hib/go-winres@${GOWINRES_VERSION}" \
      && /build/go-winres --help
 
-FROM dev-base AS containerd
+# containerd
+FROM base AS containerd-src
+WORKDIR /usr/src/containerd
+RUN git init . && git remote add origin "https://github.com/containerd/containerd.git"
+# CONTAINERD_VERSION is used to build containerd binaries, and used for the
+# integration tests. The distributed docker .deb and .rpm packages depend on a
+# separate (containerd.io) package, which may be a different version as is
+# specified here. The containerd golang package is also pinned in vendor.mod.
+# When updating the binary version you may also need to update the vendor
+# version to pick up bug fixes or new APIs, however, usually the Go packages
+# are built from a commit from the master branch.
+ARG CONTAINERD_VERSION=v1.6.10
+RUN git fetch -q --depth 1 origin "${CONTAINERD_VERSION}" +refs/tags/*:refs/tags/* && git checkout -q FETCH_HEAD
+
+FROM base AS containerd-build
+WORKDIR /go/src/github.com/containerd/containerd
 ARG DEBIAN_FRONTEND
-RUN --mount=type=cache,sharing=locked,id=moby-containerd-aptlib,target=/var/lib/apt \
-    --mount=type=cache,sharing=locked,id=moby-containerd-aptcache,target=/var/cache/apt \
-        apt-get update && apt-get install -y --no-install-recommends \
-            libbtrfs-dev
-ARG CONTAINERD_VERSION
-COPY /hack/dockerfile/install/install.sh /hack/dockerfile/install/containerd.installer /
-RUN --mount=type=cache,target=/root/.cache/go-build \
-    --mount=type=cache,target=/go/pkg/mod \
-        PREFIX=/build /install.sh containerd
+ARG TARGETPLATFORM
+RUN xx-apt-get install -y --no-install-recommends gcc libbtrfs-dev libsecret-1-dev
```

Comment:
It occurred to me that you're removing the cache mounts for apt. Is that to parallelise things, or what was the reason? Or is it that if we want to cache, we should have a unique identifier for each os/arch?

So for the apt-cache; I recall that apt does locking, and because of that we added `sharing=locked` and used a unique ID; this is also still recommended in https://github.com/moby/buildkit/blob/v0.11.0-rc1/frontend/dockerfile/docs/reference.md#example-cache-apt-packages

> Apt needs exclusive access to its data, so the caches use the option `sharing=locked`, which will make sure multiple parallel builds using the same cache mount will wait for each other and not access the same cache files at the same time. You could also use `sharing=private` if you prefer to have each build create another cache directory in this case.

Reply:
Oh I forgot to put cache mounts back but I wonder if this is still relevant as this is only useful for (local) incremental builds. From what I see, these packages are not going to change very often.

Reply:
Also previously the installer script cache was busted and therefore containerd repo needed to be cloned again if `apt-get` run changes which is not the case now as we have a dedicated mount for sources.

Reply:
The "local" incremental builds is definitely what I use daily 😅, so if the cache mounts for apt help with that, then it would definitely be nice to keep them.

(the script bust was indeed an issue)